packages/expo-updates/android/src/main/java/expo/modules/updates/db/dao/UpdateDao.java

```diff
@@ -0,0 +1,89 @@
+package expo.modules.updates.db.dao;
+
+import androidx.room.RoomWarnings;
+import androidx.room.Update;
+import expo.modules.updates.db.enums.UpdateStatus;
+import expo.modules.updates.db.entity.AssetEntity;
+import expo.modules.updates.db.entity.UpdateEntity;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import androidx.room.Dao;
+import androidx.room.Insert;
+import androidx.room.Query;
+import androidx.room.Transaction;
+
+@Dao
+public abstract class UpdateDao {
+  /**
+   * for private use only
+   * must be marked public for Room
+   * so we use the underscore to discourage use
+   */
+  @Query("SELECT * FROM updates WHERE status IN (:statuses);")
+  public abstract List<UpdateEntity> _loadUpdatesWithStatuses(List<UpdateStatus> statuses);
+
+  @Query("SELECT * FROM updates WHERE id = :id;")
+  public abstract List<UpdateEntity> _loadUpdatesWithId(UUID id);
+
+  @Query("SELECT * FROM updates INNER JOIN assets ON updates.launch_asset_id = assets.id WHERE updates.id = :id;")
```

Comment:
```suggestion
  @Query("SELECT assets.* FROM assets INNER JOIN updates ON updates.launch_asset_id = assets.id WHERE updates.id = :id;")
```

does this fix the cursor_mismatch warning?